test/functional/api/arbmark_spec.lua

```diff
@@ -0,0 +1,18 @@
+-- Sanity checks for buffer_* API calls via msgpack-rpc
```

Comment:
api/ is for RPC calls or events, this should probably be directly in functional/. Also new tests are normally written with one variable per line, not `local clear, nvim, buffer = …`. And I would suggest to never use `nvim`, `curbuf`, `curwin` and avoid `eval`: `nvim('meth', …)` is `meths.meth(…)`, same for curbuf and curwin (curbufmeths, curwinmeths), `nvim('eval', 'arbmark_index(haha)')` is `funcs.arbmark_index('haha')` (and note that you are trying to refer to variable `haha` in your code, I guess you actually wanted string `'haha'`).